### ISSUE TITLE
npm run develop gets stuck on Windows 10

### ISSUE THREAD

**Issue by rolandvar** [FoalTS/foal#1115]
# Description
The "npm run develop" command gets stuck with the message _"Starting child process with 'node ./build/index.js'"_ and the development server fails to start without showing any errors. It has always worked flawlessly in the past.

I tried on 2 windows computers to create a new Foal project and run the command "npm run develop" but in both cases it crashes.

**System information:**
Foal: 2.9.0
OS: Windows 10
NodeJS: >= v16.10.0
NPM: >= 7.24.0
MySQL: 8+

# Steps to reproduce the problem:

 1. Create a new project with the command "foal createapp my-app"
 2. Edit the "ormconfig" file to connect to a MySQL 8+ database
 3. Run the command "npm run dev"

<details>
  <summary>package.json</summary>

```json
{
  "name": "my-app",
  "version": "0.0.0",
  "description": "description of the application",
  "license": "MIT",
  "repository": {},
  "scripts": {
    "build": "foal rmdir build && tsc -p tsconfig.app.json",
    "start": "node ./build/index.js",
    "develop": "npm run build && concurrently \"tsc -p tsconfig.app.json -w\" \"supervisor -w ./build,./config -e js,json,yml --no-restart-on error ./build/index.js\"",
    "build:test": "foal rmdir build && tsc -p tsconfig.test.json",
    "start:test": "mocha --file ./build/test.js \"./build/**/*.spec.js\"",
    "test": "npm run build:test && concurrently \"tsc -p tsconfig.test.json -w\" \"mocha --file ./build/test.js -w \\\"./build/**/*.spec.js\\\"\"",
    "build:e2e": "foal rmdir build && tsc -p tsconfig.e2e.json",
    "start:e2e": "mocha --timeout 4000 --file ./build/e2e.js \"./build/e2e/**/*.js\"",
    "e2e": "npm run build:e2e && concurrently \"tsc -p tsconfig.e2e.json -w\" \"mocha --file ./build/e2e.js -w \\\"./build/e2e/**/*.js\\\"\"",
    "lint": "eslint --ext ts src",
    "lint:fix": "eslint --ext ts --fix src",
    "makemigrations": "foal rmdir build && tsc -p tsconfig.app.json && npx typeorm migration:generate --name migration && tsc -p tsconfig.app.json",
    "migrations": "npx typeorm migration:run",
    "revertmigration": "npx typeorm migration:revert"
  },
  "engines": {
    "node": ">=10"
  },
  "dependencies": {
    "@foal/core": "^2.0.0",
    "@foal/typeorm": "^2.0.0",
    "better-sqlite3": "~7.5.1",
    "mysql2": "^2.3.3",
    "source-map-support": "~0.5.19",
    "typeorm": "0.2.26"
  },
  "devDependencies": {
    "@types/mocha": "7.0.2",
    "@types/node": "10.17.24",
    "@typescript-eslint/eslint-plugin": "~2.7.0",
    "@typescript-eslint/parser": "~2.7.0",
    "concurrently": "~5.3.0",
    "eslint": "^6.7.0",
    "mocha": "~8.2.0",
    "supertest": "~6.2.3",
    "supervisor": "~0.12.0",
    "typescript": "~4.0.2"
  }
}
```
</details>

# Tests carried out

- I have tried to reproduce the problem on machines with OS Linux and MacOS but the problem does not occur.
- Used two different Windows 10 machines and the problem occurs on both;
- Attempted to update the library concurrently, but did not solve the problem;
- Tried to use different versions of node and npm and they all presented the problem;

# Investigations carried out

I tried to figure out what the problem was, I tried running "npm run develop" commands individually and they work fine.

I have found that the problem is caused by the use of "concurrently" combined with "supervisor" which for some strange reason crashes when connecting to the database.
In fact, by commenting out the line "await createConnection ();" connection to the db in the app.controller.ts file the server starts correctly.

# Possible solutions

During my research I discovered that the possible solutions to the problem are:
- Add the "-r" parameter to the "concurrently" command
- Add the parameter "-t" to the "supervisor" command
- Replace "supervisor" with "nodemon"

Am I the only one experiencing this problem?

Greetings

**Comment by scho-to** [comment]
I have got a similar Problem in Windows 11. Im still investigating for more information.

**Comment by rolandvar** [comment]
> I have got a similar Problem in Windows 11. Im still investigating for more information.
> 
> EDIT: Ok it was my mistake. I now tested "npm run develop"
> 
> * with node installed on my windows 11 pc
> * with a docker container based on the "node:16.16.0-alpine3.16"-Image. It works in both cases.

Hello,

can you try with [this repo](https://github.com/rolandvar/foal_issue_1115) ?

You have to clone it, change the connection parameters to the db in the _.env.development_ file and try the _npm run develop_ command

Greetings

**Comment by scho-to** [comment]
Hi @rolandvar,

Yeah, it also gets stuck on the same process on my windows 11 pc when running the installed node (v16.16). When I run the same in a docker container based on the node:16.16.0-alpine3.16 image, it works flawlessly.

**Comment by LoicPoullain** [comment]
Hi @rolandvar 👋 

Thank you for pointing this out and for all the work you've done to investigate the issue. 👍  And thank you @scho-to for checking this also on your side. 🙂

> During my research I discovered that the possible solutions to the problem are:
> 
>     * Add the "-r" parameter to the "concurrently" command
> 
>     * Add the parameter "-t" to the "supervisor" command
> 
>     * Replace "supervisor" with "nodemon"

Let's go with the first solution then. It looks like we also had a similar issue on a linux machine in the past and the `-r` also fixed it (#1022).

@rolandvar do you want to open a PR and add `-r` flag? The process and files to update will be the same as in #1109, https://github.com/FoalTS/foal/issues/1097#issuecomment-1195077551

**Comment by LoicPoullain** [comment]
Fixed in v2.10
